dcgtc/dgrants
# Polygon deploys and checkout

Changes:
- Deploys contracts to Polygon mainnet
- Updates app to work with Polygon
    - Configure `chains.ts` with Polygon info
    - Change `startBlock` used when scanning for events from 0 to a value specified in `chains.ts`

Notes:
- Creating a new grant redirects to the newly created grant detail page, but the page is empty for a few seconds until the data loads, but with no UI indication that data is loading. Is this a known issue or should we create an issue to track/resolve it?

To test:
- Use the information [here](https://docs.polygon.technology/docs/develop/network-details/network) to add Polygon to MetaMask
- Set `VITE_DGRANTS_CHAIN_ID=137` in `app/.env`
- Clear site data by going to the browser console > Application > Storage > Clear site data (might be a little different for Firefox). This is required because the app is only configured to work with one canonical Chain ID
- Obtain MATIC (the native token on Polygon, instead of ETH) by one of the following approaches:
    1. Use one or more of the various MATIC faucets to obtain small amounts of MATIC (enough for some transactions). Faucets: [1](https://faucet.firebird.finance/), [2](https://macncheese.finance/matic-polygon-mainnet-faucet.php), [3](https://matic.supply/) (they're a bit flaky, so try them all)
    1. Buy MATIC on L1 and [bridge](https://wallet.polygon.technology/bridge) it to Polygon 
    1. [Bridge](https://wallet.polygon.technology/bridge) ETH from L1 to MATIC, use a faucet from (i) to get MATIC, use SushiSwap to swap your bridged WETH to WMATIC, use Etherscan to unwrap WMATIC for MATIC. This is a hassle, but cheaper than the previous option if you already have L1 ETH and no L1 MATIC
    1. DM me your address and I'll send you $5 of MATIC which should last a long time (contract deploys are a few cents, everything else is typically less than a cent)
- Use SushiSwap to swap some of your MATIC fo DAI, USDC, USDT, which is the list of currently supported tokens for checkout
- When sending transactions, bump the suggested gas price a bit so it gets mined faster

TODO (things that can arguably be done in a separate PR)
- Fix estimated amount in screenshot below. There's also `Quote Update Failed` in the console, which is likely related

![image](https://user-images.githubusercontent.com/17163988/134693837-d079224f-557a-4688-82e2-d470706bc6da.png)
